lib/linebuffer.js

```diff
@@ -34,12 +34,13 @@ const needsRetry = (error, response, body) => {
 
     if (!response) { return false; }
 
-    if (response.statusCode !== 200) {
-        debug(`req failed: ${body}`);
-        if (bufferStats.flushcount % 10 === 0) { utils.log(`request failed: ${body}`); }
+    if (!response.statusCode.toString().startsWith('2')) {
```

Comment:
This check looks uncommon. To check error response code, I usually check ```if (response.statusCode >= 400```.

Reply:
Actually, you are right! I was considering `3xx` option as well but we only reply back either `2xx` or `4xx` or `5xx` in case of `Server Error`

Reply:
Updated, @weizou19!